### PR TITLE
jest-elasticsearch setup main functional

### PR DESCRIPTION
**Pull request by terret**

### CODE REVIEW

File: jest-es-config.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const documentsMapping = require('./documents-mapping');
+
+module.exports = function generateMapping(
```
Comment (vladgolubev) [inline jest-es-config.js]:
```suggestion
module.exports = function getClusterSettings(
```

File: jest-es-config.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const documentsMapping = require('./documents-mapping');
```
Comment (vladgolubev) [inline jest-es-config.js]:
```suggestion
const indexMapping = require('./index-mapping');
```

File: jest-es-config.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    indexes: [
+      {
+        name: indexName || 'documents',
+        body: {
+          settings: {
+            number_of_shards: '1',
+            number_of_replicas: '1'
+          },
+          aliases: {
+            [aliasName || 'some-doc-id']: {}
```
Comment (vladgolubev) [inline jest-es-config.js]:
```suggestion
            [aliasName]: {}
```

Defaulting here will be very unexpected for library users

File: jest-es-config.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  elasticSearchVersion
+) {
+  return {
+    esVersion: elasticSearchVersion || '6.8.2',
+    clusterName: 'docs',
+    nodeName: 'docs',
+    port: 9200,
+    indexes: [
+      {
+        name: indexName || 'documents',
```
Comment (vladgolubev) [inline jest-es-config.js]:
```suggestion
        name: indexName,
```

Defaulting here will be very unexpected for library users

File: package.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "jest-elasticsearch",
+  "version": "1.0.0",
```
Comment (vladgolubev) [inline package.json]:
```suggestion
  "version": "0.0.0",
```

1.0.0 will come a bit later)

File: package.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  },
+  "jest": {
+    "preset": "./jest-preset.js"
+  },
+  "dependencies": {
+    "@elastic/elasticsearch": "7.5.0",
+    "@shelf/quotation-marks-regex": "0.1.0"
+  },
+  "devDependencies": {
+    "@shelf/elasticsearch-local": "0.2.0",
```
Comment (vladgolubev) [inline package.json]: Looks like this is not a `dev` dependency but a prod dependency

W/o this lib jest preset won't work

File: package.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    "@shelf/eslint-config": "0.14.0",
+    "@shelf/indexing-es": "7.6.8",
```
Comment (vladgolubev) [inline package.json]: Hey, this is not a public npm package ;)

File: src/insert-documents.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+const {refreshAllIndexes} = require('./elasticsearch');
+const {getClient} = require('./elasticsearch');
```
Comment (vladgolubev) [inline src/insert-documents.js]: Why require same file 2 times? Maybe combine?

File: package.json [hunk 9]
```diff
@@ -47,7 +47,8 @@
   },
   "dependencies": {
     "@elastic/elasticsearch": "7.5.0",
-    "@shelf/quotation-marks-regex": "0.1.0"
+    "@shelf/quotation-marks-regex": "0.1.0",
```
Comment (vladgolubev) [inline package.json]: Why this dependency is needed?

File: package.json (hunk 7 above)
Comment (terret) [reply package.json]: indexing-es deleted

File: package.json (hunk 9 above)
Comment (terret) [reply package.json]: deleted

**Comment by vladgolubev** [comment]
* add yarn build
* `@elastic/elasticsearch` from prod to dev deps
* fix index.js requiring itself
* readme.md
* `jest-es-config.js` - ask people to create full file to allow full extensibility, don't accept arguments to

**Comment by terret** [comment]
**DONE:** 
- @elastic/elasticsearch from prod to dev deps
- fix index.js requiring itself
- jest-es-config.js - ask people to create full file to allow full extensibility, don't accept arguments to
  ----------------------------------------

- "yarn build" - need to be clarified
- readme.md - will be done soon

File: .circleci/config.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v2-dependencies-
+
+      - run: yarn
+
+      - save_cache:
+          paths:
+            - node_modules
+            - yarn.lock
+            - ./node_modules/.cache/mongodb-memory-server/mongodb-binaries
```
Comment (vladgolubev) [inline .circleci/config.yml]:
```suggestion
```

File: .circleci/config.yml [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+      - image: circleci/node:12
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v2-dependencies-
```
Comment (vladgolubev) [inline .circleci/config.yml]:
```suggestion
            - v1-dependencies-
```

File: .circleci/config.yml [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+      - run: yarn
+
+      - save_cache:
+          paths:
+            - node_modules
+            - yarn.lock
+            - ./node_modules/.cache/mongodb-memory-server/mongodb-binaries
+          key: v2-dependencies-{{ checksum "package.json" }}
```
Comment (vladgolubev) [inline .circleci/config.yml]:
```suggestion
          key: v1-dependencies-{{ checksum "package.json" }}
```

File: src/search-by-term.test.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
+const documents = require('./documents.mock');
+const getDocuments = require('./search-by-term');
+const insertDocuments = require('./insert-documents');
+
+describe('getDocuments', () => {
+  beforeAll(async () => {
+    await insertDocuments(documents);
+  });
+
+  it('should return wikis by term', async () => {
```
Comment (KnupMan) [inline src/search-by-term.test.js]:
```suggestion
  it('should return document by id', async () => {
```

File: src/search-by-term.test.js [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+    const wikis = await getDocuments({
```
Comment (KnupMan) [inline src/search-by-term.test.js]:
```suggestion
    const doc = await getDocuments({
```

File: src/search-by-term.test.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+      index: 'documents',
+      _id: 'some-doc-id'
+    });
+
+    expect(wikis).toEqual({
```
Comment (KnupMan) [inline src/search-by-term.test.js]:
```suggestion
    expect(docs).toEqual({
```

File: .circleci/config.yml [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+orbs:
+  node: circleci/node@1.1.6
+
+commands:
+  install_deps:
+    steps:
+      - node/with-cache:
+          cache-version: v1-all
+          cache-key: yarn.lock
```
Comment (vladgolubev) [inline .circleci/config.yml]:
```suggestion
          cache-key: package.json
```

File: src/insert-documents.js [hunk 17]
```diff
@@ -0,0 +1,31 @@
+const {refreshAllIndexes, getClient} = require('./elasticsearch');
+
+module.exports = async function insertDocuments(document) {
+  const body = [
+    {update: {_index: 'documents', _id: document.id, routing: document.id}},
```
Comment (KnupMan) [inline src/insert-documents.js]:
```suggestion
    {index: {_index: 'documents', _id: document.id, routing: document.id}},
```

File: src/index.js [hunk 18]
```diff
@@ -0,0 +1 @@
+module.exports = require('./search-by-term');
```
Comment (vladgolubev) [inline src/index.js]: What's the purpose of `index.js` and `index.test.js` ? Tests will run for the required file even without it

File: src/insert-documents.js [hunk 19]
```diff
@@ -0,0 +1,31 @@
+const {refreshAllIndexes, getClient} = require('./elasticsearch');
+
+module.exports = async function insertDocuments(document) {
```
Comment (vladgolubev) [inline src/insert-documents.js]: It says `insertDocuments` but accept only single document to insert

File: src/search-by-term.test.js [hunk 20]
```diff
@@ -0,0 +1,35 @@
+const documents = require('./documents.mock');
```
Comment (vladgolubev) [inline src/search-by-term.test.js]: File is called `documents` but exports an object of 1 document, maybe export an array?

**Pull request merged**